firewood/src/merkle.rs

```diff
@@ -120,9 +113,9 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
         Ok(if let Some(root) = root {
             let mut node = self.get_node(root)?;
             let res = node.get_root_hash::<S>(self.store.as_ref()).clone();
-            if node.lazy_dirty.load(Relaxed) {
+            if node.is_dirty() {
```

Comment:
I explicitly said that this was the case. The issue is that I moved around files to make things easier to reason about, and I did that when I already had a bunch of "branch with path" code in place. 

Again, I will state that the goal here is to finish the implementation, know that it works (priority number one), and then break down the PR into smaller PRs if and only if it makes it easier to review. 

Again, I was the first person to identify that I made the mistake of adding the path without properly spiking on all the places where it was necessary to make the changes. I had LARGE swaths of code that were in a half-working state when I realized that I needed to refactor some of this other code to make it easier to grok. I continued to refactor as I went further down the rabbit hole, with the explicit priority of solving my initial goal. A lot of this refactoring was necessary because of floating constants that only made sense based on some other code in some random place. As I was trying to solve the task at hand, it became unwieldy to traverse up and down the large files. It was also easier to edit multiple files at the same time when they were split up.

Reply:
I know. I'm just marking those areas that can use this refactor, and why it's just a nit.